test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/PauseUnpauseTokenAccountPrecompileSuite.java

```diff
@@ -0,0 +1,350 @@
+import static com.hederahashgraph.api.proto.java.TokenPauseStatus.Paused;
+import static com.hederahashgraph.api.proto.java.TokenPauseStatus.Unpaused;
+import static com.hederahashgraph.api.proto.java.TokenType.FUNGIBLE_COMMON;
+import static com.hederahashgraph.api.proto.java.TokenType.NON_FUNGIBLE_UNIQUE;
+
+import com.hedera.services.bdd.spec.HapiApiSpec;
+import com.hedera.services.bdd.suites.HapiApiSuite;
+import com.hederahashgraph.api.proto.java.TokenID;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class PauseUnpauseTokenAccountPrecompileSuite extends HapiApiSuite {
+    private static final Logger log =
+            LogManager.getLogger(PauseUnpauseTokenAccountPrecompileSuite.class);
+    private static final String PAUSE_UNPAUSE_CONTRACT = "PauseUnpauseTokenAccount";
+
+    private static final String PAUSE_KEY = "PAUSE_KEY";
+
+    private static final String UNPAUSE_KEY = "UNPAUSE_KEY";
+
+    private static final String ACCOUNT = "account";
+
+    public static final long INITIAL_BALANCE = 1_000_000_000L;
+    private static final long GAS_TO_OFFER = 4_000_000L;
+    private static final String PAUSE_TOKEN_ACCOUNT_FUNCTION_NAME = "pauseTokenAccount";
+    private static final String UNPAUSE_TOKEN_ACCOUNT_FUNCTION_NAME = "unpauseTokenAccount";
+    private static final String PAUSE_FUNGIBLE_TXN = "pauseFungibleTxn";
+    private static final String UNPAUSE_FUNGIBLE_TXN = "unpauseFungibleTxn";
+    private static final String PAUSE_NONFUNGIBLE_TXN = "pauseNonFungibleTxn";
+    private static final String UNPAUSE_NONFUNGIBLE_TXN = "unpauseNonFungibleTxn";
+
+    public static void main(String... args) {
+        new PauseUnpauseTokenAccountPrecompileSuite().runSuiteSync();
+    }
+
+    @Override
+    protected Logger getResultsLogger() {
+        return log;
+    }
+
+    @Override
+    public List<HapiApiSpec> getSpecsInSuite() {
+        return allOf(
+                List.of(
+                        pauseFungibleTokenHappyPath(),
+                        unpauseFungibleTokenHappyPath(),
+                        pauseNonFungibleTokenHappyPath(),
+                        unpauseNonFungibleTokenHappyPath()));
+    }
+
+    private HapiApiSpec pauseFungibleTokenHappyPath() {
+        final AtomicReference<TokenID> tokenID = new AtomicReference<>();
+
+        return defaultHapiSpec("PauseFungibleTokenHappyPath")
+                .given(
+                        newKeyNamed(PAUSE_KEY),
+                        cryptoCreate(TOKEN_TREASURY),
+                        cryptoCreate(ACCOUNT).balance(INITIAL_BALANCE),
+                        tokenCreate(VANILLA_TOKEN)
+                                .tokenType(FUNGIBLE_COMMON)
+                                .treasury(TOKEN_TREASURY)
+                                .pauseKey(PAUSE_KEY)
+                                .adminKey(PAUSE_KEY)
```

Comment:
The naming here is a bit confusing. We have `PAUSE_KEY` set to two different types of keys. Probably you can name this key `MULTI_PURPOSE_KEY` or assign `ADMIN_KEY` as admin key.